gems/aws-sdk-core/lib/aws-sdk-core/client_stubs.rb

```diff
@@ -262,13 +262,17 @@ def apply_stubs(operation_name, stubs)
     end
 
     def convert_stub(operation_name, stub)
-      case stub
+      stub = case stub
       when Proc then stub
       when Exception, Class then { error: stub }
       when String then service_error_stub(stub)
       when Hash then http_response_stub(operation_name, stub)
       else { data: stub }
       end
+      if Hash === stub
+        stub[:mutex] = Mutex.new
+      end
```

Comment:
Why does this need to be separate from the Hash case?

Reply:
Other cases (everything except the Proc) produce a hash, which we then need to add the mutex to.
With a proc, it gets called at operation time to produce a new hash, which doesn't need the mutex.